android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/TabListReducer.kt

```diff
@@ -211,8 +211,9 @@ internal object TabListReducer {
                 state.copy(
                     tabs = normalTabs,
                     selectedTabId = if (selectionAffected) {
-                        // If the selection is affected, select the last normal tab, if available.
-                        normalTabs.lastOrNull()?.id
+                        // If the selection is affected, we'll set it to null as there's no
+                        // normal tab left and NO normal tab should get selected instead.
+                        null
```

Comment:
I don't recall this change in my first pass through. Could you provide some additional scenario and context for this change?

Reply:
Sure! I discovered the `TabListAction.RemoveAllPrivateTabsAction` either through a test or maybe just auto-complete. I figured it would be worth ensuring behavior matched across the two actions. This is used, for example, by the notification that is active while in private mode to [close all private tabs](https://searchfox.org/mozilla-mobile/rev/99012f8a4534bbd3b29e3fe73764baf0a27fdb24/firefox-android/android-components/components/feature/privatemode/src/main/java/mozilla/components/feature/privatemode/notification/AbstractPrivateNotificationService.kt#76)

Reply:
Okay, got it! I like to think the fact that we were adding extra tests and having a smaller PR here helped us discover more blindspots.